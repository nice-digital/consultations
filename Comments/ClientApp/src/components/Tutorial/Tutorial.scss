@use "@nice-digital/nds-core/scss/colours";
@use "@nice-digital/nds-core/scss/utils";
@use "@nice-digital/nds-core/scss/media-queries";
@use "../../styles/variables" as *;

.tutorial {
  background-color: $ec-light-grey;

  &__button-image {
    display: block;
    width: 75%;
    margin: 0 auto utils.rem(20px);

    @include media-queries.mq($from: sm) {
      width: 80%;
      margin: 0;
    }
  }

  &__video {
    position: relative;
    width: 100%;
    overflow: hidden;
    padding-top: 80%; /* weird aspect ratio */

    iframe {
      position: absolute;
      top: 0;
      left: 0;
      bottom: 0;
      right: 0;
      width: 100%;
      height: 100%;
      border: none;
    }
  }
}
