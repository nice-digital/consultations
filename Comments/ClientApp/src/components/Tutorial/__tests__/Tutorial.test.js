import React from "react";
import { render } from "@testing-library/react";
import { Tutorial } from "../Tutorial";
import Cookies from "js-cookie";

test("should match snapshot and display the tutorial when no cookie is set", () => {
	const {container} = render(<Tutorial />);
	expect(container).toMatchSnapshot();
});

test("should match snapshot and display the tutorial when no cookie is set to true", () => {
	Cookies.get = jest.fn().mockImplementation(() => "true");
	const {container} = render(<Tutorial />);
	expect(container).toMatchSnapshot();
});

test("should match snapshot and display the tutorial when no cookie is set to false", () => {
	Cookies.get = jest.fn().mockImplementation(() => "false");
	const {container} = render(<Tutorial />);
	expect(container).toMatchSnapshot();
});
