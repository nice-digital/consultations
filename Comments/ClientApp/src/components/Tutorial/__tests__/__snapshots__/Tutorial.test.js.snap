// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot and display the tutorial when no cookie is set 1`] = `
<div>
  <aside
    class="tutorial mt--0 pt--d pb--d"
  >
    <div
      class="container"
    >
      <button
        class="buttonAsLink"
      >
        Hide how to comment
      </button>
      <div
        class="grid"
      >
        <div
          data-g="12 sm:7"
        >
          <span
            class="h1 pt--c"
          >
            How to comment
          </span>
          <p>
            Please view our 5-minute video on how to use the online commenting system.
          </p>
          <p>
            See how to:
          </p>
          <ul>
            <li>
              Create comments and respond to questions
            </li>
            <li>
              Add, edit and delete comments
            </li>
            <li>
              Create comments specific to sections of the document
            </li>
            <li>
              Review all your feedback before submitting your response
            </li>
          </ul>
        </div>
        <div
          data-g="12 sm:5"
        >
          <div
            class="tutorial__video"
          >
            <iframe
              allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
              allowfullscreen=""
              frameborder="0"
              src="https://www.youtube.com/embed/DILcZkxVPCY?rel=0"
              title="YouTube video player"
            />
          </div>
        </div>
      </div>
    </div>
  </aside>
</div>
`;

exports[`should match snapshot and display the tutorial when no cookie is set to false 1`] = `
<div>
  <aside
    class="tutorial mt--0 pt--d pb--d"
  >
    <div
      class="container"
    >
      <button
        class="buttonAsLink"
      >
        Show how to comment
      </button>
    </div>
  </aside>
</div>
`;

exports[`should match snapshot and display the tutorial when no cookie is set to true 1`] = `
<div>
  <aside
    class="tutorial mt--0 pt--d pb--d"
  >
    <div
      class="container"
    >
      <button
        class="buttonAsLink"
      >
        Hide how to comment
      </button>
      <div
        class="grid"
      >
        <div
          data-g="12 sm:7"
        >
          <span
            class="h1 pt--c"
          >
            How to comment
          </span>
          <p>
            Please view our 5-minute video on how to use the online commenting system.
          </p>
          <p>
            See how to:
          </p>
          <ul>
            <li>
              Create comments and respond to questions
            </li>
            <li>
              Add, edit and delete comments
            </li>
            <li>
              Create comments specific to sections of the document
            </li>
            <li>
              Review all your feedback before submitting your response
            </li>
          </ul>
        </div>
        <div
          data-g="12 sm:5"
        >
          <div
            class="tutorial__video"
          >
            <iframe
              allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
              allowfullscreen=""
              frameborder="0"
              src="https://www.youtube.com/embed/DILcZkxVPCY?rel=0"
              title="YouTube video player"
            />
          </div>
        </div>
      </div>
    </div>
  </aside>
</div>
`;
